github-connector/routing.go

```diff
@@ -8,21 +8,44 @@ import (
 	"github.com/google/go-github/github"
 )
 
-func handleWebhook(w http.ResponseWriter, r *http.Request) {
-	payload, err := github.ValidatePayload(r, []byte("my-secret-key"))
+
+//Validator is an interface used to allow mocking the github library methods
+type Validator interface {
+	ValidatePayload(*http.Request, []byte) ([]byte, error)
+	ParseWebHook(string, []byte) (interface{}, error)
+	GetToken() string
+}
+
+//WebHookHandler is a struct used to allow mocking the github library methods
+type WebHookHandler struct {
+	validator Validator
+}
+
+//NewWebHookHandler creates a new webhook handler with the passed interface
+func NewWebHookHandler(v Validator) *WebHookHandler {
+	return &WebHookHandler{validator: v}
+}
+
+func (wh *WebHookHandler) handleWebhook(w http.ResponseWriter, r *http.Request) {
+
+	payload, err := wh.validator.ValidatePayload(r, []byte(wh.validator.GetToken()))
+
 	if err != nil {
 		log.Printf("error validating request body: err=%s\n", err)
+		w.WriteHeader(http.StatusUnauthorized)
```

Comment:
The header itself gives only information on which area did the error occur in. The good practice would be to create some kind of unified error response for your component - containing more precise information on the error that appeared during request processing.

This stays the same for all of the "unhappy path" cases - BadRequests, etc.